vita3k/app/src/app_init.cpp

```diff
@@ -101,11 +102,86 @@ void update_viewport(EmuEnvState &state) {
     }
 }
 
+void init_paths(Root &root_paths) {
+    const auto dir_sep = std::string{ fs::path::preferred_separator };
+    auto base_path = SDL_GetBasePath();
+    auto pref_path = SDL_GetPrefPath(org_name, app_name);
+    root_paths.set_base_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_pref_path(string_utils::utf_to_wide(pref_path));
+    root_paths.set_log_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_config_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_shared_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_cache_path(fs::path(string_utils::utf_to_wide(base_path)) / "cache" / dir_sep);
+    SDL_free(base_path);
+    SDL_free(pref_path);
+
+#if defined(__linux__) && !defined(__ANDROID__) && !defined(__APPLE__)
+    // XDG Data Dirs.
+    auto env_home = getenv("HOME");
+    auto env_data = getenv("XDG_DATA_HOME");
+    auto env_home_data = env_data;
+    if (env_data != NULL)
+        root_paths.set_pref_path(fs::path(string_utils::utf_to_wide(env_data)) / app_name / app_name / dir_sep);
```

Comment:
the utf_to_wide function is applied to path because of windows, but this code is running on linux only, you can get rid of the calls and pass env_data directly to fs::path